src/main/java/org/radarcns/android/auth/LoginActivity.java

```diff
@@ -133,6 +133,10 @@ public void run() {
     /** Call when the entire login procedure succeeded. */
     public void loginSucceeded(LoginManager manager, @NonNull AppAuthState appAuthState) {
         logger.info("Login succeeded");
+
+        if (!this.appAuth.isValid() && appAuthState.isValid()) {
+            this.appAuth = appAuthState;
```

Comment:
Well if both are invalid it doesn't really matter, and this way it will not change the behavior if `appAuth` was valid before.

I guess a different way would be to have getter/setter for `appAuth` and removing the passed `appAuthState` altogether. This way one could set it from the Login Manager before calling `loginSucceeded`.

Reply:
My initial idea was that this would indeed behave like a setter, plus some extensible logic. If there was a login from one user to another, both appauth might be valid, but we want to take the most recent one. Then perhaps
```java
if (appAuthState.isValid()) {
    this.appAuth = appAuthState;
}
```

Reply:
OK yeah, i think this would be the best way to do it.